cloud/pkg/synccontroller/objectsync.go

```diff
@@ -78,8 +80,15 @@ func sendEvents(err error, nodeName string, sync *v1alpha1.ObjectSync, resourceT
 	if err != nil && apierrors.IsNotFound(err) {
 		//trigger the delete event
 		klog.Infof("%s: %s has been deleted in K8s, send the delete event to edge", resourceType, sync.Spec.ObjectName)
-		msg := buildEdgeControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.DeleteOperation, obj)
-		beehiveContext.Send(commonconst.DefaultContextSendModuleName, *msg)
+		source := strings.Split(sync.Labels["source"], ",")
+		if IsExist(source, modules.DynamicControllerModuleName) {
+			msg := buildDynamicControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.DeleteOperation, obj)
+			beehiveContext.Send(commonconst.DefaultContextSendModuleName, *msg)
+		}
+		if IsExist(source, modules.EdgeControllerModuleName) {
+			msg := buildEdgeControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.DeleteOperation, obj)
+			beehiveContext.Send(commonconst.DefaultContextSendModuleName, *msg)
```

Comment:
`buildDynamicControllerMessage` and `buildEdgeControllerMessage`  may return nil, which should be handled for robustness purposes